server/handlers_linkpreview.go

```diff
@@ -39,6 +39,31 @@ func getThumbnailPayload(db *sql.DB, logger *zap.Logger, msgID string, thumbnail
 	return payload, nil
 }
 
+func getFaviconPayload(db *sql.DB, logger *zap.Logger, msgID string, thumbnailURL string) ([]byte, error) {
```

Comment:
I don't think we should duplicate this much. This function is 99% identical to `getThumbnailPayload`, except for the part `payload = p.FaviconPayload`.